util/sqlexec/restricted_sql_executor.go

```diff
@@ -34,3 +34,10 @@ type RestrictedSQLExecutor interface {
 	// ExecRestrictedSQL run sql statement in ctx with some restriction.
 	ExecRestrictedSQL(ctx context.Context, sql string) (ast.RecordSet, error)
 }
+
+// SQLParser is a bit tricky. We could run parser.New() to get a parser object, and then run Parse method on it.
+// But a session already has a parser bind in it, so we define this interface to avoid allocate a new parser.
+// See session.SQLParser for more information.
+type SQLParser interface {
```

Comment:
We should first describe the function of this interface. But the comment only contains the detail of its implementation.